File: InsertionSort.java
/*Java program for Insertion Sort*/

import java.util.Scanner;

public class InsertionSort {
	static int n;//size of array
	static int a[];//array of numbers to be sorted
	//Function to perform Insertion sort
	public static void sort() {
		//Inserting elements while sorting
				for (int i = 1; i < n; ++i)  
				{ 
				//Taking next element
				 int key = a[i]; 
				 int j = i-1; 
				 //Comparing existing elements with new element
				 while (j>=0 && a[j]>key)  
				  { 
				  a[j + 1] = a[j]; //Finding correct place for the new element
				  j = j - 1; 
				  } 
				 a[j + 1] = key; 
				}	
	}
	//Function to take user input
		public static void userInput() {
			Scanner sc=new Scanner(System.in);
			System.out.println("Enter size");
			n=sc.nextInt();
			a=new int[n];
			System.out.println("Enter numbers to sort:");
			for(int i=0;i<n;i++) {
				a[i]=sc.nextInt();
			}
		}
		//Function to print output
		public static void printOutput() {
			System.out.println("The sorted order is:");
			for(int i=0;i<n;i++) {
				System.out.println(a[i]);
			}
		}
	public static void main(String args[]) {
		userInput();//Taking user input
		sort();//Call to sort the array
		printOutput();//Printing output
		
	}
	
}

File: cpp/insertion_sort/insertion_sort.cpp
#include <iostream>
using namespace std;
void swap(int &a,int &b){
    int c=a;
    a=b;
    b=c;
}

void insertion_sort(int arr[],int len){
     
     int temp,j;
     for(int i=1;i<len;i++){
         temp=arr[i];
         j=i;
         while(j>0&&arr[j-1]>temp){
             arr[i]=arr[j-1];
             j--;
         }
         arr[j]=temp;
     }
     cout<<"The sorted array using insertion sort is  ";
     for(int i=0;i<len;i++){
         cout<<arr[i]<<" ";
     }
}
int main() {
    int n;
    cout<<"enter no of elements in array ";
    cin >>n;
    int arr[n];
    cout<<"enter elements in the array ";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    insertion_sort(arr,n);

    return 0;
}
